pandas/io/formats/csvs.py

```diff
@@ -263,23 +263,21 @@ def save(self) -> None:
             self._save()
 
         finally:
-            if self.should_close:
-                f.close()
-            elif (
-                isinstance(f, TextIOWrapper)
-                and not f.closed
-                and f != self.path_or_buf
-                and hasattr(self.path_or_buf, "write")
-            ):
+            if handleArgs.is_wrapped:
                 # get_handle uses TextIOWrapper for non-binary handles. TextIOWrapper
                 # closes the wrapped handle if it is not detached.
-                f.flush()  # make sure everything is written
-                f.detach()  # makes f unusable
-                del f
-            elif f != self.path_or_buf:
-                f.close()
-            for _fh in handles:
-                _fh.close()
+                assert isinstance(handleArgs.handle, TextIOWrapper)
+                handleArgs.handle.flush()
+                handleArgs.handle.detach()
+                handleArgs.created_handles.remove(handleArgs.handle)
+            for handle in handleArgs.created_handles:
```

Comment:
it would be *really* nice to push this logic to IOHandleArgs itself in a method, not sure how easy this would be

Reply:
I like that idea a lot. Something like `IOHandleArgs.close()` to take care of everything :)

Reply:
I also added `IOArgs.close()`